iree/hal/cts/allocator_test.cc

```diff
@@ -0,0 +1,83 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "iree/base/status.h"
+#include "iree/hal/cts/cts_test_base.h"
+#include "iree/hal/testing/driver_registry.h"
+#include "iree/testing/gtest.h"
+#include "iree/testing/status_matchers.h"
+
+namespace iree {
+namespace hal {
+namespace cts {
+
+class AllocatorTest : public CtsTestBase {};
+
+// Tests for baseline buffer compatibility that all HAL drivers must support.
+TEST_P(AllocatorTest, QueryBufferCompatibility) {
```

Comment:
these aren't really true - not supporting mapping is fine (for example over RPC channels to other machines), and not supporting device local memory is fine too - the only thing we do need is either host local/device visible *or* host visible/device local (as otherwise there's no way to get data between the host and device) but mapping on either is optional

Reply:
Cool, that's the feedback I was looking for!

I expanded coverage in the test and made it more specific about what it is checking for and why. How does it look now?